### ISSUE TITLE
Posting a toot that's translation of another toot

### ISSUE THREAD

**Issue by Changaco** [tootsuite/mastodon#6920]
It would be great if it was possible to "tag" a toot as being a translation of another toot, because it would allow frontends to "collapse" toots which say the same thing in different languages into a single message.

In the UI a new menu item labeled "Add a translation" could be added above "Pin on profile", like this:

![Screenshot of a toot's dropdown with an extra option labeled "Add a translation"](https://user-images.githubusercontent.com/1581590/37905832-7aaca130-3100-11e8-9109-83d19ecf46de.png)

Clicking it would be like replying to the toot, but the new toot wouldn't be a reply, it would be a separate toot with metadata indicating that it's a translation. It could look like this:

![Screenshot of the compose area when replying to a toot, modified to display the text "Adding a translation of:" instead of a link to the original toot's author](https://user-images.githubusercontent.com/1581590/37906710-45fea07a-3103-11e8-92da-636b5a401fcd.png)

I think this proposal is a little different than the previously rejected idea #821, but I'll understand if it's also deemed to be too complex.

* * * *

- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.

**Comment by m4sk1n** [comment]
https://www.w3.org/TR/activitypub/#i18n-concerns

**Comment by Wolf480pl** [comment]
@m4sk1n 
>language detection
umm... how about let the user select which language they're posting in?

**Comment by Changaco** [comment]
@Wolf480pl That's a separate issue: #3478.

**Comment by Changaco** [comment]
In case there are concerns that multilingual features could end up cluttering the UI, they could be hidden by default behind an account setting, such as a checkbox labeled "I toot in multiple languages".

**Comment by Changaco** [comment]
I'm not really familiar with Activity Streams but from glancing at the specs they don't seem to support translations in the same way they support replies, i.e. there is an [`inReplyTo` property](https://www.w3.org/TR/activitystreams-vocabulary/#dfn-inreplyto) but no `isTranslationOf`.

**Comment by Changaco** [comment]
I've opened an issue about that: <https://github.com/w3c/activitystreams/issues/463>.

**Comment by comradekingu** [comment]
Maybe using https://hosted.weblate.org/projects/liberapay integrates well with the workflow.
Translations are sent as git commits almost instantly, so it only needs a import/export function to the mastadon accounts/mirrorsmoke.

**Comment by Changaco** [comment]
No, Weblate isn't the right tool for this job. It would make it difficult to handle images, for example.

**Comment by comradekingu** [comment]
@Changaco Unless images are embedded, as opposed to links, I don't see why not?

**Comment by nightpool** [comment]
I think this feature is too complicated for a microblogging platform. I also don't understand how this would be useful above and beyond just adding a normal reply.

**Comment by Changaco** [comment]
A translation is not a reply, it's an alternate version of the same message. If translations were posted as replies then followers who hide replies in their Home timeline would never see any translation, while followers who don't hide replies would keep seeing every message.

**Comment by Changaco** [comment]
> I think this feature is too complicated for a microblogging platform.

The complexity of multilingualism is unavoidable. Either Mastodon will handle it or users will have to do it themselves.

If Mastodon isn't improved then we'll probably have to create new accounts for every language we toot in, and set up a bot to automatically boost toots from our main multilingual account into the language-specific accounts. I would rather not have to spend time setting up and maintaining such a system.

**Comment by Cassolotl** [comment]
No one minds when you toot in more than one language though, I think? I happily follow many people who toot in multiple languages.

**Comment by Changaco** [comment]
Individuals usually have a primary language they toot in, and they don't translate all their toots into multiple languages. That's why you don't mind: you don't get lots of duplicate messages in languages you can't read.

We've only tooted in 3 languages so far and several of our followers have already suggested that we need multiple accounts to avoid flooding their timelines with translations. Imagine if we tooted in the ~17 languages that https://liberapay.com/ is currently available in…
